File: sdk/src/main/java/com/adadapted/android/sdk/SessionAdapter.java
package com.adadapted.android.sdk;

import org.json.JSONObject;

/**
 * Created by chrisweeden on 3/26/15.
 */
interface SessionAdapter {
    interface Listener {
        void onSessionRequestCompleted(JSONObject response);
    }

    void sendInit(JSONObject json);

    void addListener(Listener listener);
    void removeListener(Listener listener);
    void notifySessionRequestCompleted(JSONObject response);
}

File: advertising_sdk/src/main/java/com/adadapted/android/sdk/core/interfaces/SessionAdapter.kt
package com.adadapted.android.sdk.core.interfaces

import com.adadapted.android.sdk.core.device.DeviceInfo
import com.adadapted.android.sdk.core.session.Session
import com.adadapted.android.sdk.core.view.ZoneContext

interface SessionAdapter {
    suspend fun sendInit(deviceInfo: DeviceInfo, listener: SessionInitListener)
    suspend fun sendRefreshAds(session: Session, listener: AdGetListener, zoneContexts: MutableSet<ZoneContext>)
}